### ISSUE TITLE
Self-signed certificate notice is being displayed without the certificate being generated

### ISSUE THREAD

**Issue by madonius** [matrix-org/synapse#4620]
### Description
The notice about the self signed certificates is being displayed after the configuration file generation although this was removed in #4509 
See: https://github.com/matrix-org/synapse/blob/3bd9daf4b86ed811c9ced95a4adb9aa58b681399/synapse/config/_base.py#L316

### Steps to reproduce

Generate the configuration file

<!-- 
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**:  local test environment

<!-- 
What version of Synapse is running? 
You can find the Synapse version by inspecting the server headers (replace matrix.org with
your own homeserver domain):
$ curl -v https://matrix.org/_matrix/client/versions 2>&1 | grep "Server:"
-->
- **Version**: `v0.99.0`

- **Install method**: Docker
<!-- examples: package manager/git clone/pip  -->

- **Platform**:  Docker
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

**Comment by richvdh** [comment]
I think this will be fixed by #4619

**Comment by madonius** [comment]
This is not regarding the docker-specific templating. Rather we just used docker to generate a config file, we used 
```bash
python -m synapse.app.homeserver \
    --server-name my.domain.name \
    --config-path homeserver.yaml \
    --generate-config \
    --report-stats=no
```
As described in https://github.com/matrix-org/synapse/blob/master/INSTALL.md

**Comment by madonius** [comment]
cc @jcgruenhage

**Comment by richvdh** [comment]
in that case, can you give more details about the repro steps and what symptoms you actually see?

**Comment by jcgruenhage** [comment]
```bash
jcgruenhage  ~  docker run -it --rm --entrypoint /bin/sh docker.io/matrixdotorg/synapse
/ # python -m synapse.app.homeserver \
>     --server-name my.domain.name \
>     --config-path homeserver.yaml \
>     --generate-config \
>     --report-stats=no
A config file has been generated in 'homeserver.yaml' for server name 'my.domain.name' with corresponding SSL keys and self-signed certificates. Please review this file and customise it to your needs.
If this server name is incorrect, you will need to regenerate the SSL certificates
```

**Comment by richvdh** [comment]
Right, the warning we are complaining about is this one:

```
 jcgruenhage  ~  docker run -it --rm --entrypoint /bin/sh docker.io/matrixdotorg/synapse
/ # python -m synapse.app.homeserver \
>     --server-name my.domain.name \
>     --config-path homeserver.yaml \
>     --generate-config \
>     --report-stats=no
A config file has been generated in 'homeserver.yaml' for server name 'my.domain.name' with corresponding SSL keys and self-signed certificates. Please review this file and customise it to your needs.
If this server name is incorrect, you will need to regenerate the SSL certificates
```

**Comment by neilisfragile** [comment]
Looks fixed by #4625
